content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,170 @@
+---
+title: "How to run workflows using reana-client Python API"
+date: 2021-02-22T15:00:00+01:00
+tags:
+  - API
+  - client
+  - Python
+---
+
+Here you can find a step by step guide on how to submit workflows to
+REANA using the `reana-client` Python API. This is a simple example,
+for illustrative purposes, using the serial workflow engine and a
+simple [ROOT6 / RooFit analysis](https://github.com/reanahub/reana-demo-root6-roofit/) example.
+We will create a new workflow, run it and retrieve its results using
+the `reana-client` Python API.
+<!--more-->
+
+Install `reana-client`:
+```
+$ pip install reana-client==0.7.0
+```
+
+Folowing instructions can be executeed inside a Python interpreter,
+or a Python file.
+
+First we need to provide the minimum configuration:
+- Point the REANA service URL:
+```python
+# with `reana-client` it would be
+#   `export REANA_SERVER_URL=https://reana.cern.ch`
+import os
+
+os.environ['REANA_SERVER_URL'] = 'https://reana.cern.ch'
+```
+
+- Provide our REANA access token:
+```python
+# with `reana-client` it would be
+#  `export REANA_ACCESS_TOKEN=XXXXXXX`
+from getpass import getpass
+
+my_reana_token = \
+  os.getenv("REANA_ACCESS_TOKEN") or getpass('Enter your REANA token: ')
+```
+
+Now we will choose a name for the workflow and the kind of
+workflow specification we will use:
+```python
+my_workflow_name = 'root6-roofit'
+workflow_type = 'serial'
+```
+
+Let us specify inputs and workflows specification:
+- So first we will provide it's inputs: (files and parameters)
```

Comment:
```suggestion
- So first we will provide its inputs: (files and parameters)
```

Reply:
These two parenthesis blocks separated with the colon look a bit strange to me, maybe we can display the info a bit differently?

Reply:
Meta-comment: do not use itemised lists, we can use text paragraphs and code blocks. Also, the text paragraphs can be much expanded to use a lot of verbiage, in a tutorial style, so that the text would be highly readable. One example: instead of the very brief "Point the REANA service URL", we can say something very explicit like: "Let us set up the environment variable REANA_SERVER_URL. This variable is mandatory and should point to the remote REANA cluster instance that will be used to run our computations. The client code will use this environment variable automatically to connect to the remote REANA cluster later on. The typical value is https://reana.cern.ch for CERN deployments".  You know what I mean: basically the blog post should read like a small novel :wink: and be fully self-explanatory.

Reply:
On the same theme: instead of having "with `reana-client` it would be..." as a Python code comment, this text could be moved to the main narrative, expressing it more verbally, for example "This is equivalent to creating the workflow using `reana-client create` command" with link to the docs for the create command.